### PR TITLE
Fix ATS tab status indicators

### PR DESCRIPTION
**Pull request by Don-Vito**
1. Fix progress value not updated
2. Introduce TabStatus object and bind both TabHeaderControl and CommandPalette to it
3. Add support for read-only mode indicator

### CODE REVIEW

**Comment by Don-Vito** [comment]
@zadjii-msft - I missed the progress value in the original PR 😢 

Instead of an interface I creates a TabStatus object and bind all controls to it.

Marked as draft, as for some reason the ATS bindings are sometimes not updated.

File: src/cascadia/TerminalApp/TabPaletteItem.idl [hunk 1]
```diff
@@ -12,10 +13,6 @@ namespace TerminalApp
 
         TabBase Tab { get; };
 
-        Boolean IsProgressRingActive { get; set; };
-        Boolean IsProgressRingIndeterminate { get; set; };
-        UInt32 ProgressValue { get; set; };
-        Boolean IsPaneZoomed { get; set; };
-        Boolean BellIndicator { get; set; };
+        TerminalTabStatus TabStatus { get; };
```
Comment (zadjii-msft) [inline src/cascadia/TerminalApp/TabPaletteItem.idl:16]: Ah. I suppose I figured that `TabStatus` would be an interface, so you'd have something like

```
runtimeclass TabPaletteItem : PaletteItem, ITabStatus
```

which would force `TabPaletteItem` to implement all of the `IsProgressRingActive` etc declared in `ITabStatus`.

THAT BEING SAID,

This implementation does actually seem simpler to me. It's so darn convenient to just bind them both to one singular object.

File: src/cascadia/TerminalApp/TerminalTab.cpp [hunk 2]
```diff
@@ -579,25 +580,25 @@ namespace winrt::TerminalApp::implementation
...
-                    tab->IsProgressRingActive(true);
+                    tab->_tabStatus.IsProgressRingActive(true);
                 }
                 else
                 {
                     // Show the tab icon
                     tab->HideIcon(false);
-                    tab->_headerControl.IsProgressRingActive(false);
-                    tab->IsProgressRingActive(false);
+                    tab->_tabStatus.IsProgressRingActive(false);
```
Comment (zadjii-msft) [inline src/cascadia/TerminalApp/TerminalTab.cpp:601]: gosh this is so much more nicer than updating each value twice

File: src/cascadia/TerminalApp/TerminalTab.cpp [hunk 3]
```diff
@@ -676,7 +677,7 @@ namespace winrt::TerminalApp::implementation
                 }
                 tab->_focusState = WUX::FocusState::Programmatic;
                 // This tab has gained focus, remove the bell indicator if it is active
-                if (tab->_headerControl.BellIndicator())
+                if (tab->TabStatus().BellIndicator())
```
Comment (zadjii-msft) [inline src/cascadia/TerminalApp/TerminalTab.cpp:680]: Curious, this one uses `TabStatus()` but the others use `_tabStatus`. I'm assumuing there's not _really_ a reason, right? (I don't mind, I guess I'm just curious if there _is_ a reason)

**Review by zadjii-msft — COMMENTED** [review]
Yea so far I'm cool with what's here.

File: src/cascadia/TerminalApp/TerminalTab.cpp (hunk 3 above)
Comment (Don-Vito) [reply src/cascadia/TerminalApp/TerminalTab.cpp:680]: No reason, just lack of focus. I will fix it. :blush:

File: src/cascadia/TerminalApp/TabPaletteItem.idl (hunk 1 above)
Comment (Don-Vito) [reply src/cascadia/TerminalApp/TabPaletteItem.idl:16]: I started with the interface, but it still required the TabPaletteItem to register each property manually. So I switched to this.

**Comment by Don-Vito** [comment]
@zadjii-msft - for some reasons nested bindings were not updating  to me.. so I added a code in the TabPaletteItem that notifies changes on the entire status if one of the indicators changes. This should have no implications, but it is definitely annoying and feels like voodoo.

**Comment by Don-Vito** [comment]
@carlos-zamora - please take a look as well. There are some gaps I missed originally + I did something simpler to avoid properties duplication by now.

**Review by carlos-zamora — APPROVED** [review]
omg this is so much cleaner. Thank you!

**Review by zadjii-msft — APPROVED** [review]
:shipit:

**Comment by zadjii-msft** [comment]
Thanks for hoping on this so quickly. What a fun bit of polish ☺️

**Pull request merged**